scipy/signal/spectral.py

```diff
@@ -697,36 +1262,35 @@ def _spectral_helper(x, y, fs=1.0, window='hann', nperseg=256,
     References
     ----------
     .. [1] Stack Overflow, "Rolling window for 1D arrays in Numpy?",
-        http://stackoverflow.com/a/6811241
-    .. [2] Stack Overflow, "Using strides for an efficient moving average
-        filter", http://stackoverflow.com/a/4947453
+           http://stackoverflow.com/a/6811241
+    .. [2] Stack Overflow, "Using strides for an efficient moving
+           average filter", http://stackoverflow.com/a/4947453
 
     Notes
     -----
     Adapted from matplotlib.mlab
 
     .. versionadded:: 0.16.0
     """
-    if mode not in ['psd', 'complex', 'magnitude', 'angle', 'phase']:
```

Comment:
Phase derivates were never calculated by one of the modes, but can be derived either from the output of `stft` or `spectrogram` with `phase/angle` arguments.

In any case, for compatibility, I've restored the `mode` kwarg in `spectrogram`. It isn't really a thing of beauty, but it produces correct results for the first time.